packages/client/src/devices/InputMediaDeviceManager.ts

```diff
@@ -394,17 +395,21 @@ export abstract class InputMediaDeviceManager<
     }
     if (this.state.mediaStream !== stream) {
       this.state.setMediaStream(stream, await rootStream);
+      const handleTrackEnded = async () => {
+        await this.statusChangeSettled();
+        if (this.enabled) {
+          this.isTrackStoppedDueToTrackEnd = true;
+          setTimeout(() => {
+            this.isTrackStoppedDueToTrackEnd = false;
+          }, 2000);
+          await this.disable();
+        }
+      };
       this.getTracks().forEach((track) => {
-        track.addEventListener('ended', async () => {
-          await this.statusChangeSettled();
-          if (this.enabled) {
-            this.isTrackStoppedDueToTrackEnd = true;
-            setTimeout(() => {
-              this.isTrackStoppedDueToTrackEnd = false;
-            }, 2000);
-            await this.disable();
-          }
-        });
+        track.addEventListener('ended', handleTrackEnded);
+        this.subscriptions.push(() =>
+          track.removeEventListener('ended', handleTrackEnded),
+        );
```

Comment:
I think this was a possible memory leaked, fixed by removing event listener on dispose.